equinox/nn/_attention.py

```diff
@@ -152,6 +169,21 @@ def __init__(
         dropout_p: float = 0.0,
         inference: bool = False,
         dtype=None,
+        kv_cache: Optional[
+            Callable[
+                [
+                    State,
+                    Float[Array, "seq_length num_heads qk_size"],
+                    Float[Array, "seq_length num_heads vo_size"],
+                ],
+                tuple[
+                    Float[Array, "state_length num_heads qk_size"],
+                    Float[Array, "state_length num_heads vo_size"],
+                    Int[Array, ""],
+                    State,
+                ],
+            ]
+        ] = None,
```

Comment:
Oh gosh, this is probably going to make our documentation unreadable.

I don't love it, but maybe we should switch this out with just `Optional[Callable]` and then explain that it should match the same API as `StandardKVCache`?